raiden/network/resolver/client.py

```diff
@@ -41,16 +42,31 @@ def reveal_secret_with_resolver(
         "expiration": secret_request_event.expiration,
         "payment_recipient": to_hex(raiden.address),
         "chain_id": chain_state.chain_id,
-        "chain_height": chain_state.block_number,
     }
 
-    try:
-        response = requests.post(raiden.config["resolver_endpoint"], json=request)
-    except requests.exceptions.RequestException:
-        return False
+    # loop until we get a valid response from the resolver or until timeout
+    while True:
+        current_state = raiden.wal.state_manager.current_state
+        assert isinstance(current_state, ChainState)
 
-    if response.status_code != HTTPStatus.OK:
-        return False
+        if secret_request_event.expiration < current_state.block_number:
+            return False
+
+        response = None
+
+        try:
+            # before calling resolver, update block height
+            request["chain_height"] = chain_state.block_number
+            response = requests.post(raiden.config["resolver_endpoint"], json=request)
+        except requests.exceptions.RequestException:
+            pass
+
+        if response is not None:
+            if response.status_code == HTTPStatus.OK:
+                break
+            if response.status_code == HTTPStatus.NOT_FOUND:
+                return False
+        time.sleep(5)
```

Comment:
You should use `gevent.sleep` here to context-switch and allow other gevent greenlets to run while this waits. Otherwise, this will block everything.

Reply:
Also, the value of 5 seems arbitrary. This should implement exponential backoff so that less requests are made over time.

Reply:
@rakanalh I can do backoff here. However, there must be a max value for the backoff. After internal discussion we consider 5 second to be a reasonable max value. 
To simplify things, instead of starting with 0.1s and doing exponential backoff all the way to 5 seconds we decided to make it a fixed 5 second. 
Let me know still think that a change is needed.